### PR TITLE
Fixes #2265 - Broken hosts_and_facts detection

### PR DESCRIPTION
**Pull request by taysu**
Fixes this issue submitted by me:

http://projects.theforeman.org/issues/2265

Tested with Puppet master running Puppet 3.1.1.

### CODE REVIEW

**Comment by GregSutcliffe** [comment]
[test]

File: lib/tasks/puppet.rake [hunk 1]
```diff
@@ -45,7 +45,7 @@ namespace :puppet do
   namespace :import do
     desc "Imports hosts and facts from existings YAML files, use dir= to override default directory"
     task :hosts_and_facts => :environment do
-      dir = ENV['dir'] || "#{Puppet[:vardir]}/yaml/facts"
+      dir = ENV['dir'] || "#{SETTINGS[:puppetvardir]}/yaml/facts"
```
Comment (ohadlevy) [inline lib/tasks/puppet.rake:48]: didnt we changed it to Setting['puppetvardir'] ?

Comment (domcleal) [inline lib/tasks/puppet.rake:48]: no, not that I can see

**Comment by domcleal** [comment]
:+1:

**Comment by ohadlevy** [comment]
merged - thanks @andytayloruk !

**Pull request closed without merging**